miso-web/src/main/webapp/scripts/list_storage_location.js

```diff
@@ -80,11 +80,12 @@ ListTarget.storage_location = {
         mData: "retired",
         include: true,
         mRender: function (data, type, full) {
-          if (type === "display") {
-            if (full.outOfService) {
-              return "Out of Service";
+          if (full.outOfService) {
+            return "Out of Service";
+          } else {
+            if (type === "display") {
```

Comment:
This `type` parameter is telling you why DataTables (table library we're using to create these lists) is calling the `mRender` function. `display` means to get the value to display in the table. The other options are related to sorting, filtering, and figuring out the datatype for the column.

Since you're checking for out of service before checking `type`, this will end up returning a string for out of service freezers, and a boolean for others, which may cause errors or strange behaviour. Since this will always display a string, I think it should be fine to use the same string for other purposes too. You can remove the `if (type === "display")` check and always return "Out of Service", "Retired", or "Production"